src/components/ViewServiceDetails/ViewServiceDetails.js

```diff
@@ -69,54 +60,55 @@ class ViewServiceDetails extends React.Component {
 		);
 	}
 
-	getVehiclesList(){
+	getVehiclesList() {
 		var _this = this;
-		fetch('http://localhost:4001/users/5a86de0b90d792bccf3c3404/vehicles', 
-		{ 
-			method: 'GET', 
-			headers: {
-				'Content-Type': 'application/json'
-			}
-		}).then(function(response){
-			return( response.text() );
-		}).then(function(response){
-			return JSON.parse(response);
-		}).then(function(response){
-			console.log('getVehiclesList response', response.data);
-			_this.setState({
-				vehicles: response.data
+		fetch('http://localhost:4001/users/5a86de0b90d792bccf3c3404/vehicles',
+			{
+				method: 'GET',
+				headers: {
+					'Content-Type': 'application/json'
+				}
+			}).then(function (response) {
+				return (response.text());
+			}).then(function (response) {
+				return JSON.parse(response);
+			}).then(function (response) {
+				console.log('getVehiclesList response', response.data);
+				_this.setState({
+					vehicles: response.data || []
+				});//setting empty array to prevent crash if reponse is undefined
```

Comment:
The code is self explanatory here. Comments are needed only when it is complex.

Reply:
removed the comments